dev/NavigationView/NavigationView.cpp

```diff
@@ -397,7 +397,8 @@ void NavigationView::CreateAndHookEventsToSettings(std::wstring_view settingsNam
         bool shouldSelectSetting = selectedItem && IsSettingsItem(selectedItem);
 
         if (shouldSelectSetting)
-        { 
+        {
+            m_shouldIgnoreNextSelectionChange = true;
```

Comment:
Please follow this pattern:
```
                auto scopeGuard = gsl::finally([this]()
                {
                    m_shouldIgnoreNextSelectionChange = false;
                });
                m_shouldIgnoreNextSelectionChange = true;
```
Otherwise some functionality may break.

Reply:
Although below also works, but scopeGuard is preferred.
```
m_shouldIgnoreNextSelectionChange = true;
DoSomething
m_shouldIgnoreNextSelectionChange = false;
```

Reply:
I changed the code in question to use the scopeGuard as suggested.